src/main/resources/com/puppycrawl/tools/checkstyle/grammars/javadoc/JavadocLexer.g4

```diff
@@ -120,10 +120,10 @@ Newline5: NEWLINE
 Leading_asterisk3: LEADING_ASTERISK -> type(LEADING_ASTERISK);
 XmlTagOpen1: '<' -> type(OPEN), pushMode(xmlTagDefinition);
 STRING: '"' .*? '"' {referenceCatched = false;} -> mode(DEFAULT_MODE);
-PACKAGE: [a-z_$] ([a-z0-9_$] | '.')+ [a-z0-9_$] {referenceCatched = true;};
+PACKAGE_CLASS: [a-zA-Z_$] ('.'? [a-zA-Z0-9_$]+)* {referenceCatched = true;};
 DOT: '.';
 HASH: '#' {referenceCatched = true;} -> mode(classMemeber);
-CLASS: [A-Z] [a-zA-Z0-9_$]* {referenceCatched = true;};
+CLASS: [a-zA-Z_$] [a-zA-Z0-9_$]* {referenceCatched = true;};
```

Comment:
Please confirm if this change is fine here or should we create a separate commit (probably a `minor: `) for this.